### ISSUE TITLE
[a11y]: Tabbed content does not have an associated name

### ISSUE THREAD

**Issue by danielannan17** [carbon-design-system/carbon#9666]
### Package

carbon-components-react

### Browser

Firefox

### Operating System

MacOS

### Package version

v7.43.0

### Automated testing tool and ruleset

IBM Equal Access Accessbility Checker

### Description

Components with the ARIA role are expected to have an associated name

### CodeSandbox example

https://8u88r.csb.app/

### Steps to reproduce

1. Download IBM Equal Access Accessibility Checker firefox extension
2. Visit https://8u88r.csb.app/
3. Open chrome dev tools
4. Navigate to 'Accessibility Assessment' tab
5. Click 'Scan' - you should now see the violation
<img width="548" alt="Screenshot 2021-09-15 at 08 55 12" src="https://user-images.githubusercontent.com/11006124/133396372-d16d8df5-e749-4226-b0b0-641e05e3f56e.png">

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by abbeyhrt** [comment]
Hi @danielannan17 👋  Thanks for making this issue!  Tabbed content is aria-labelled by `Tab` so `Tab` needs to have an `id` for it to correctly label the tab content. You can see this in the code sandbox you sent, if you add an `id` to `Tab` the accessibility violation goes away. [Example with id added](https://codesandbox.io/embed/eager-wildflower-4meuw?fontsize=14&hidenavigation=1&theme=dark)

**Comment by abbeyhrt** [comment]
Closing since the issue has been addressed.
